pkg/route/generator/generate.go

```diff
@@ -77,8 +78,14 @@ func (RouteGenerator) Generate(genericParams map[string]interface{}) (runtime.Ob
 
 	portString := params["target-port"]
 	if len(portString) > 0 {
+		var targetPort intstr.IntOrString
+		if port, err := strconv.Atoi(portString); err == nil {
+			targetPort = intstr.FromInt(port)
+		} else {
+			targetPort = intstr.FromString(portString)
+		}
 		route.Spec.Port = &api.RoutePort{
-			TargetPort: intstr.FromString(portString),
+			TargetPort: targetPort,
```

Comment:
Yes, `IntValue` returns a number independent of what `portString` is. I would have to check if that number is zero in order to determine if `portString` is a number or a string. Instead, by doing the conversion directly I know on the spot what I need to use (`FromInt` vs `FromString`).